src/main/java/me/botsko/prism/utils/MaterialTag.java

```diff
@@ -1,202 +1,250 @@
+    }
+
+    /**
+     * Constructor.
+     * add  Material via matching.
+     * @param segment Sting
+     * @param mode MatchMode
+     * @return MaterialTag
+     */
+    public MaterialTag(String segment, MatchMode mode) {
+        this.materials = EnumSet.noneOf(Material.class);
+        append(segment, mode);
+    }
+
+    @NotNull
+    @Override
+    public NamespacedKey getKey() {
+        return key;
+    }
+
+    public MaterialTag append(Material... materials) {
+        this.materials.addAll(Arrays.asList(materials));
+        return this;
+    }
+
+    /**
+     * add new Tags to the group.
+     * @param materialTags Tag
+     * @return MaterialTag
+     */
+    @SafeVarargs
+    public final MaterialTag append(Tag<Material>... materialTags) {
+        for (Tag<Material> materialTag : materialTags) {
+            this.materials.addAll(materialTag.getValues());
+        }
+
+        return this;
+    }
+
+    /**
+     * Add a material by name with a match mode.
+     * @param segment String
+     * @param mode MatchMode
+     * @return MaterialTag
+     */
+    public MaterialTag append(String segment, MatchMode mode) {
+        segment = segment.toUpperCase();
+
+        switch (mode) {
+            case PREFIX:
+                for (Material m : Material.values()) {
+                    if (m.name().startsWith(segment)) {
+                        materials.add(m);
+                    }
+                }
+                break;
+
+            case SUFFIX:
+                for (Material m : Material.values()) {
+                    if (m.name().endsWith(segment)) {
+                        materials.add(m);
+                    }
+                }
+                break;
+
+            case CONTAINS:
+                for (Material m : Material.values()) {
+                    if (m.name().contains(segment)) {
+                        materials.add(m);
+                    }
+                }
+                break;
+            default:
+                throw new IllegalArgumentException(mode.name() + " is NOT a valid rule");
+        }
+
+        return this;
+    }
+
+    /**
+     * Exclude materials
+     * @param materials Materials
+     * @return  MaterialTag
+     */
+    public MaterialTag exclude(Material... materials) {
+        for (Material m : materials) {
+            this.materials.remove(m);
+        }
+
+        return this;
+    }
+
+    /**
+     * Exclude Tags from this group.
+     * @param materialTags Tag
+     * @return MaterialTag
+     */
+    @SafeVarargs
+    public final MaterialTag exclude(Tag<Material>... materialTags) {
+        for (Tag<Material> materialTag : materialTags) {
+            this.materials.removeAll(materialTag.getValues());
+        }
+
+        return this;
+    }
+    /**
+     * Exclude Tags from this group.
+     * @param segment String
+     * @param mode MatchMode
+     * @return MaterialTag
+     */
+    public MaterialTag exclude(String segment, MatchMode mode) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['METHOD_DEF' should be separated from previous statement.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)